### ISSUE TITLE
[Bug] Enable Plugin to Properly Acknowledge Props Provided Via `React.FC`

### ISSUE THREAD

**Issue by ITenthusiasm** [yannickcr/eslint-plugin-react#3045]
Carryover from #2353, intended to remove the noise from the problem.

As stated in the opening comment of the original issue, this plugin does not properly validate props when `React.FC` is used. Information was provided at typescript-eslint/typescript-eslint#3075 on how to address this issue. The suggested approach just needs to be implemented.

cc: @ljharb 

## Original Opening Comment from @dchebakov 
**Tell us about your environment**

- ESLint Version: v6.0.1
- Node Version: v10.16.0
- npm Version: v6.9.0

**Configuration**

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json',
  },
  plugins: ['@typescript-eslint'],
  extends: [
    'plugin:react/recommended',
    'plugin:@typescript-eslint/recommended',
  ],
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```tsx
import * as React from 'react';
interface PersonProps {
    username: string;
}
const Person: React.FunctionComponent<PersonProps> = (props): React.ReactElement => (
    <div>{props.username}</div>
);
```

```bash
npx eslint src --ext .tsx
```

**What did you expect to happen?**
`eslint` show no errors, because type definition actually present in generic type `React.FunctionComponent<PersonProps>`

**What actually happened? Please include the actual, raw output from ESLint.**

```js
error    'username' is missing in props validation       react/prop-types
```
